File: src/fractal/world/fractum/quantum/_state.py
"""
Quantum State
=============

An enum of possible discrete states for a quantum.

# [TODO] dont like Python enum, any way around?
# [TODO] make ordering a classification

"""

from abc import ABCMeta


class QuantumStateMeta:
    def __new__(cls, attr):
        pass

    def __getattribute__(self, attr: str):
        return QuantumState(attr)


class QuantumState(
    int,
    metaclass=QuantumStateMeta,
):
    pass


class Test:
    class TicTacToe_QuantumState(
        QuantumState,
    ):
        Empty = 0
        White = 1
        Black = 2

File: src/fractal/world/fractum/quantum/_state.rs
//! An enum of possible discrete states for a quantum.
//!
//! TODO redesign this with Rust
//!
//!<TODO> dont like Python enum, any way around?
//!<TODO> make ordering a pub structification




pub struct QuantumStateMeta:
    fn __new__(cls, attr):
        pass

    fn __getattribute__(&self, attr: str):
        return QuantumState(attr)


pub struct QuantumState(
    i32,
    metapub struct=QuantumStateMeta,
):
    pass


pub struct Test:
    pub struct TicTacToe_QuantumState(
        QuantumState,
    ):
        Empty = 0
        White = 1
        Black = 2
